### ISSUE TITLE
`too_long_first_doc_paragraph` ignores indentation

### ISSUE THREAD

**Issue by clarfonthey** [rust-lang/rust-clippy#13448]
### Summary

This lint ignores indentation in its auto-fixes, meaning it creates poorly formatted code.

### Reproducer

```rust
#![warn(clippy::too_long_first_doc_paragraph)]

pub struct Thing;
impl Thing {
    /// This is a one-line summary.
    /// This makes the one-line summary no longer one line.
    pub fn method() {}
}
```

autofixes to:

```rust
#![warn(clippy::too_long_first_doc_paragraph)]

pub struct Thing;
impl Thing {
    /// This is a one-line summary.
    ///
/// This makes the one-line summary no longer one line.
    pub fn method() {}
}
```

### Version

rustc 1.83.0-nightly (7042c269c 2024-09-23)
binary: rustc
commit-hash: 7042c269c166191cd5d8daf0409890903df7af57
commit-date: 2024-09-23
host: x86_64-unknown-linux-gnu
release: 1.83.0-nightly
LLVM version: 19.1.0

**Comment by SpriteOvO** [comment]
Hmmm, wait, is this a new lint request? I can't find anything about the lint `too_long_first_doc_comment` that you mentioned, and clippy also reports it's a unknown lint.

**Comment by SpriteOvO** [comment]
I can't reproduce the fixing bug, the lint currently doesn't handle items in impl block at all, I'm fixing this problem now.

**Comment by SpriteOvO** [comment]
The name of the lint suggests that the lint does not distinguish between item types (i.e., impl-block and trait-block should be covered as well), even though the purpose of the lint is to prevent the rendering of excessively long summaries.

@GuillaumeGomez What is your opinion on this? Should we cover all items?

**Comment by GuillaumeGomez** [comment]
The idea for this lint is to prevent having too long "short docs", which you can only see on module pages. There isn't really a point for items that don't show up in the module pages imo.

**Comment by clarfonthey** [comment]
I think that there is value to having a succinct first paragraph, but I agree that in most cases it doesn't matter here. Regardless, it definitely shouldn't autofix while ignoring indentation.

**Comment by SpriteOvO** [comment]
Okay, I got it.

Reproducible

```rust
pub mod other {
    /// A very short summary.
    /// A much longer explanation that goes into a lot more detail about
    /// how the thing works, possibly with doclinks and so one,
    /// and probably spanning a many rows. Blablabla, it needs to be over
    /// 200 characters so I needed to write something longeeeeeeer.
    pub fn method() {}
}
```

fix to 

```rust
pub mod other {
    /// A very short summary.
    ///
/// A much longer explanation that goes into a lot more detail about
    /// how the thing works, possibly with doclinks and so one,
    /// and probably spanning a many rows. Blablabla, it needs to be over
    /// 200 characters so I needed to write something longeeeeeeer.
    pub fn method() {}
}
```

I will be making a PR fix it later today.

**Comment by SpriteOvO** [comment]
This bug is already fixed in PR #13377, and is no longer reproducible in the latest nightly.

**Comment by GuillaumeGomez** [comment]
Nice! Closing then.
